src/main/java/uk/gov/pay/api/resources/PaymentsResource.java

```diff
@@ -139,27 +109,35 @@ public PaymentsResource(Client client, ObjectMapper objectMapper, PublicApiConfi
             @ApiResponse(code = 404, message = "Not found", response = PaymentError.class),
             @ApiResponse(code = 500, message = "Downstream system error", response = PaymentError.class)})
     public Response getPayment(@ApiParam(value = "accountId", hidden = true) @Auth Account account,
-                               @PathParam(PAYMENT_KEY) String paymentId) {
+                               @PathParam("paymentId") String paymentId) {
 
         logger.info("Payment request - paymentId={}", paymentId);
+        String url = account.getPaymentType().equals(DIRECT_DEBIT) ? connectorDDUrl : connectorUrl;
+        UriBuilder builder = UriBuilder
+                .fromPath(url)
+                .path(format("/v1/api/accounts/%s/charges/%s", account.getAccountId(), paymentId));
```

Comment:
this might be for a later PR, but i was wondering if this could be covered by a connector client or connector url generator too (similar to publicApiUriGenerator).